Carbon.Cryptography/Functions/New-CRsaKeyPair.ps1

```diff
@@ -0,0 +1,375 @@
+
+function New-CRsaKeyPair
+{
+    <#
+    .SYNOPSIS
+    Generates a public/private RSA key pair.
+
+    .DESCRIPTION
+    The `New-CRsaKeyPair` function uses the `certreq.exe` program to generate an RSA public/private key pair suitable

```

Comment:
How should we indicate this function is Windows only? Docs? Write an error when not `$IsWindows`?

Reply:
Line 213:
```powershell
        $certReqPath = Get-Command -Name 'certreq.exe' | Select-Object -ExpandProperty 'Path'
        if( -not $certReqPath )
        {
            return
        }
```

Note that `Get-Command` doesn't have an `ErrorAction` parameter, so if the `certreq.exe` command doesn't exist, the user will get an error message that `certreq.exe` doesn't exist.

Reply:
I changed it to this:

```powershell
        $certReqPath = Get-Command -Name 'certreq.exe' -ErrorAction Ignore | Select-Object -ExpandProperty 'Path'
        if( -not $certReqPath )
        {
            'Command "certreq.exe" does not exist. This is a Windows-only command. If you''re on Windows, make sure ' +
            '"C:\Windows\System32" is part of your "Path" environment variable.' |
                Write-Error -ErrorAction $ErrorActionPreference
            return
        }
```